### DISCUSSION TITLE
Export notes in CSV

### DISCUSSION THREAD

**Discussion by wolfgangkarall** [iSoron/uhabits#1567] [Feature Requests]
Hello,

while testing things for #1566 I noticed the entered notes are in the sqlite3 database, but not exported in the Checkmarks.csv files. Please add a column 'notes' there as well.

And thanks for the great app. :smiley:

Cheers
Wolfgang

**Comment by k11m1** [comment]
I want to express my interest in this "feature" - if we don't call it a bug that it's not getting exported (that's where i searched at first). It would be much helpful for me to put keywords into the notes for later processing.

**Comment by 0ldb34r** [comment]
I agree. I don't understand the use of comments if i can't export them in the end.

**Comment by vnagel** [comment]
+1. The notes are an integral part of my habit entries, and so currently important context is missing when viewing the CSV in an external program.

**Comment by Nikomaes** [comment]
Hi everyone,
I hope you are alright 
I want to add my concern to this issue and ask if someone can add a feature so that the daily comments of the habits in the app are included to the export csv of them.
I believe it would help a lot specially for users of the app like me that usually want to make some comments on different days regarding the situation or caracteristics of the habits of that day and it's a bit tedious to pass all those notes manually when changing the device or creating an export for backup purposes or so on.
For the structure I've seen on the csv files the app generated it seems it wouldn't be so complicated to add a column to the checkmarks file where for each day appart from the information of the progress (either numerical or yes no) there would be a column with the message the user has written for that habit that day if there was one or perhaps a space or whatever neutral character you use to indicate it or perhaps leave it without text if there wasn't a comment in certain days but it would be of really help for many people that use the app to include in the csv the messages we write each day so if any developer or a volunteer or whoever can do it can help to add that feature to the app (specially allowing to add the previous existing comments) it would be really nice.
Thank you in advance for everything
Best wishes fot you all with blessings and success
Regards
Nic

**Comment by McBaumwolle** [comment]
I had the same issue, but then ended up using [SQLiteBrowser](https://sqlitebrowser.org/) and converted the data to '.csv' - worked like a charm!

**Reply by Julianoe** [reply]
Problem with relying on SQLiteBrowser .csv export is that you loose the human readable link between the habits and the repetitions. The Repetition.csv only contains the ID of the habit.

**Comment by Julianoe** [comment]
this single feature lacking from the backup is what makes me hesitant to continue to use the app. I heavily rely on notes to track things. But I also deem very important to be able to port my data out from the apps I use (I'm thinking long term). Having to open the SQLite file to then have the notes whilst it could simply be a column in a .csv export is a bit annoying.
+1 on that feature improvement of the backup option.

**Comment by fictiontoreality** [comment]
It seems including notes in the CSV export will be in the next release: https://github.com/iSoron/uhabits/discussions/2183#discussioncomment-14012633

**Reply by Julianoe** [reply]
Awesome ! this is a game changer for me
